server/src/main/java/com/lunchtime/service/impl/PersonServiceImpl.java

```diff
@@ -91,6 +92,18 @@ public PersonDto getPersonDtoByEmail(String email) {
         return personRepository.findById(id);
     }
 
+    @Override
+    public void updatePersonRoleId(Restaurant restaurant) {
+        Optional<Person> res = personRepository.findById(restaurant.getPersonId());
+        if (res.isPresent()) {
+            Person person = res.get();
+            if (person.getRoleId() != 2L) {
+                person.setRoleId(2L);
+            }
+            personRepository.save(person);
```

Comment:
Better. You try to fix redundant updating `role`. But in you new code you save `person` even if it isn't changed